otter/convergence/service.py

```diff
@@ -581,6 +607,23 @@ def converge(tenant_id, group_id, dirty_flag):
     yield do_return(parallel(effs))
 
 
+@do
+def get_recently_converged_groups(recently_converged, interval):
+    """
+    Return a list of recently converged groups, and garbage-collect any groups
+    in the recently_converged map that are no longer 'recent'.
+    """
+    # STM would be cool but this is synchronous so whatever
+    recent = orig = yield recently_converged.read()
+    now = yield Effect(Func(time.time))
+    for group in recent.keys():
+        if now - recent[group] > interval:
+            recent = recent.remove(group)
```

Comment:
@manishtomar the important thing to realize is that there is no mutation going on. Those are pmaps, and .remove() returns a new pmap. notice the `recent = ` part, on the line with the call to `.remove(group)`. The only "mutation" is the rebinding of the `recent` name.